src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,181 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(head, value, tail);
```

Comment:
When creating a new `Node<T>`, all the fields should be passed to the `Node` constructor. However, the `head` and `tail` references are not correctly used here. A new node should only be linked to the current tail if it exists, not to both head and tail.